Orange/widgets/data/oweditdomain.py

```diff
@@ -1555,6 +1565,18 @@ def get_data(self):
             tr.insert(0, self.format_cb.currentData())
         return var, tr
 
+    def _on_custom_change(self):
+        self.custom_text = self.custom_edit.text()
+        date_pat = r"%(-?)d|%(b|B)|%(-?)m|%(y|Y)|%(-?)j|%(-?)U|%(-?)W|%(a|A)|%w"
+        time_pat = r"%(-?)H|%(-?)I|%p|%(-?)M|%(-?)S|%f"
+        pattern1 = re.compile(date_pat)
+        pattern2 = re.compile(time_pat)
```

Comment:
Compiling if out of fashion. You can just use `re.search(pattern1, self.custom_text)` (if you keep re's at all). We use to compile them because otherwise `re.search` would compile them anew at each call. But Python now caches the compiled re, so compilation is no longer necessary.